buildenv/jenkins/jobs/infrastructure/copyrightCheck

```diff
@@ -64,24 +72,308 @@ timeout(time: 6, unit: 'HOURS') {
+                                        regexsElm = regexsElm + ".*"
+                                    }
+                                } else {
+                                    regexsElm = regexsElm.replaceAll("[/]","");
+                                }
+                            }
+                            if (regexsElm.contains ("/+.*.*")) {
+                                regexsElm = regexsElm.replace("/+.*.*","/*.*.*");
+                            }
+                            
+                            if (regexsElm ==~ /.*.[A-Za-z0-9]/){
+                                print regexsElm;
+                                regexsElm = regexsElm.substring(0, regexsElm.lastIndexOf(".")) + /\./ + regexsElm.substring(regexsElm.lastIndexOf(".") + 1);
+                            }
+                            
+                            REGEXLIST << regexsElm
+                        }
+                        print REGEXLIST
+                        FILES_LIST.each { fileName ->
+                           for (regex in REGEXLIST) {
+                                if (fileName ==~ /${regex}/) {
+                                    FILES_LIST = FILES_LIST.collect { it.toString() } - "${fileName}"
+                                    echo "Ignoring File: '${fileName}', appears to be in WHITELIST, matching " + /${regex}/
+                                    break
+                                }
+                            }
+                        }
+                    }
+
+                    FILES_LIST.each { file ->
+                        def checkCopyrightDate = true
+                        if (ghprbGhRepository ==~ "ibmruntimes.*") {
+                            println "Checking file: '${file}'"
+                            
+                            def lineCount = 0
+                            def foundOracleCopyright = -1
+                            def foundIBMCopyright = -1
+                            def foundIBMPortionsCopyright = -1
+                            def foundGPLv2 = -1
+                            def foundCPE = -1, foundIBMCPE = -1, foundOracleCPE = -1;
+                            def foundBSDorMITCopyright = -1
+                            def foundApacheCopyright = -1
+                            def foundCopyright = -1
+                            def errorStr
+                            def foundNonIBMCopyright = false
+                            def inJDK = true
+                            
+                            def theFile = readFile "${file}"
+                            
+                            if( theFile.length() <= 0 ) {
+                                println "File is empty"
+                            } else {
+                                String[] fileLines = theFile.split("\n")
+                                toLines = fileLines.size()
+                                if (toLines > 100) { toLines = 100 }
```

Comment:
Have we verified all the existing copyrights we have occur within the first 100 lines?

Reply:
Most are. This was taken from Peter Shipton's java program he wrote to check the copyright.